### PR TITLE
[WKWebKit] Update to Xcode 9.3 Beta 1

### PR DESCRIPTION
**Pull request by chamons**
- All of the various NSSecureCoding were already handled by our code
- https://github.com/xamarin/xamarin-macios/wiki/WebKit-iOS-xcode9.3-beta1 and https://github.com/xamarin/xamarin-macios/wiki/WebKit-macOS-xcode9.3-beta1

### CODE REVIEW

File: src/wkwebkit.cs [hunk 1]
```diff
@@ -435,6 +443,10 @@ interface WKWebsiteDataStore : NSSecureCoding {
 	interface WKOpenPanelParameters	{
 		[Export ("allowsMultipleSelection")]
 		bool AllowsMultipleSelection { get; }
+
+		[Mac (10, 13, 4)]
```
Comment (spouliot) [inline src/wkwebkit.cs]: I don't recall if inheritance covers this but it's easier when reviewing if the `onlyOn64 : true` is added on members (even if the type has it).

File: tests/xtro-sharpie/macOS-WebKit.todo [hunk 2]
```diff
@@ -1,10 +1,2 @@
-!missing-field! WKWebsiteDataTypeFetchCache not bound
-!missing-field! WKWebsiteDataTypeServiceWorkerRegistrations not bound
 !missing-protocol-conformance! WebView should conform to NSUserInterfaceValidations (defined in 'WebIBActions' category)
```
Comment (spouliot) [inline tests/xtro-sharpie/macOS-WebKit.todo]: it this something fixed in master ?
or does it need something else in foundation (NS prefix) ?

File: tests/xtro-sharpie/macOS-WebKit.todo [hunk 3]
```diff
@@ -1,10 +1,2 @@
...
-!missing-protocol-conformance! WKPreferences should conform to NSSecureCoding
-!missing-protocol-conformance! WKProcessPool should conform to NSSecureCoding
-!missing-protocol-conformance! WKUserContentController should conform to NSSecureCoding
-!missing-protocol-conformance! WKWebsiteDataStore should conform to NSSecureCoding
 !missing-protocol-conformance! WKWebView should conform to NSUserInterfaceValidations (defined in 'WKIBActions' category)
```
Comment (spouliot) [inline tests/xtro-sharpie/macOS-WebKit.todo]: same

File: src/wkwebkit.cs [hunk 4]
```diff
@@ -392,6 +392,14 @@ interface WKWebsiteDataType {
 		
 		[Field ("WKWebsiteDataTypeIndexedDBDatabases", "WebKit")]
 		NSString IndexedDBDatabases { get; }
+
+		[Mac (10, 13, 4), iOS (11, 3)]
```
Comment (spouliot) [inline src/wkwebkit.cs]: `onlyOn64 : true`

File: src/wkwebkit.cs [hunk 5]
```diff
@@ -392,6 +392,14 @@ interface WKWebsiteDataType {
...
+		[Field ("WKWebsiteDataTypeFetchCache")]
+		NSString FetchCache { get; }
+
+		[Mac (10, 13, 4), iOS (11, 3)]
```
Comment (spouliot) [inline src/wkwebkit.cs]: `onlyOn64 : true`

File: tests/xtro-sharpie/macOS-WebKit.todo (hunk 2 above)
Comment (chamons) [reply tests/xtro-sharpie/macOS-WebKit.todo]: Completely unsure. It wasn't in the web headers so I assumed it'd come up in Foundation later. I can dig more now if needed, but assumed if it's in todo it's ok.

**Comment by chamons** [comment]
We had this: 
```	[Availability (Introduced = Platform.iOS_8_0, Deprecated = Platform.iOS_9_0, Message = "Use 'RequiresUserActionForMediaPlayback' or 'MediaTypesRequiringUserActionForPlayback' instead.")]
```
I was waffling on if I should update it to remove the first part or not.

**Comment by chamons** [comment]
build

**Comment by spouliot** [comment]
@chamons I think that's fine. IIRC they have different availability so it make sense to show both (to avoid warning twice or showing an unusable alternative)

**Comment by dalexsoto** [comment]
issuing new build so we get xtro report.

**Comment by dalexsoto** [comment]
build

**Pull request merged**